woocommerce/woocommerce-gateway-stripe
# Adds statement previews for full and shortened bank statement text.

# Changes proposed in this Pull Request:

Fixes #1792

This creates a new StatementPreview component that accepts `icon`, `title`, `text`, and `className` props. The icon should be a string that matches a pre-defined property name of an icon SVG in the component. Current options for this are `creditCard` and `bank`. The `title` is a string of text shown at the top of the preview and describes the preview. `text` is the text that will show under the "Transaction" heading on the preview. And `className` will be added to the outer `div` of the preview card.

The "amount" shown on the card will respect the shop's currency settings for currency symbol, currency symbol position, decimal separator, and precision.

I've also added two tests to ensure that the text content from the boxes above appears in the preview cards when rendered.

![statement-previews-min](https://user-images.githubusercontent.com/1558827/133614461-44b95016-a3c2-4805-b569-826d2eb239c2.gif)

# Testing instructions

* Ensure you have the `_wcstripe_feature_upe_settings` flag enabled
* Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings
* The "All Payment Methods" preview card is rendered with the text from the "Customer bank statement" field above.
* Check the "Add customer order number to the bank statement" box.
* The "Cards & Express Checkouts" preview card is rendered with the text from the "Shortened customer bank statement" field above with "* #123456" appended to it
